### DISCUSSION TITLE
HA Bluetooth Integration Preparation

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#719] [Announcements]
### Beings Bluetooth is all the rage now...
I decided to create a script from the info at https://www.home-assistant.io/integrations/bluetooth/#requirements-for-linux-systems

#### The script will ask for permission before each step. 

It adds the bullseye-backport sources.
Installs and enables dbus-broker and switches from dbus-daemon
Installs BlueZ 

Run in the Home Assistant LXC console.
```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/bluetooth.sh)"
```
* I use Home Assistant Core and experienced performance improvements using dbus-broker.
* I currently don't use BlueZ, I'm using [BT Proxy](https://esphome.github.io/bluetooth-proxies/) instead of a USB device.

**Comment by tteck** [comment]
The [dbus-broker project](https://github.com/bus1/dbus-broker) is an implementation of a message bus as defined by the D-Bus specification. Its aim is to provide high performance and reliability, while keeping compatibility to the D-Bus reference implementation. It is exclusively written for Linux systems, and makes use of many modern features provided by recent Linux kernel releases.

**Comment by bkemmler** [comment]
Is Bluetooth working for HA with the BT Proxy in a LXC? HA always tells me not Bluetooth :-(

**Reply by tteck** [reply]
Works great! 
BT Proxy uses ESPHome, Home Assistant will auto-discovery the proxies. 

![Screenshot 2022-11-26 6 54 34 PM](https://user-images.githubusercontent.com/21342670/204113241-edd4bb47-4164-4730-92fb-ac64cf3a95de.png)
![Screenshot 2022-11-26 6 55 36 PM](https://user-images.githubusercontent.com/21342670/204113244-9f694bec-3493-4a68-b2db-e6a1b237e5bd.png)

**Comment by webtoucher** [comment]
Do you know any way of proxying Bluetooth without external hardware?

**Reply by tteck** [reply]
I don't

**Comment by thealphonso** [comment]
Hello!
Tried running this within HA terminal, not from HA LXC console and got this:
![image](https://user-images.githubusercontent.com/22868370/235534601-51bf9f89-7a9f-46e4-9dab-13d58edfc29d.png)
When trying to run it from HA LXC console, got this:
![image](https://user-images.githubusercontent.com/22868370/235534773-2ebc0849-bae4-4905-a316-6fb3a187155e.png)
LE: I'm not running LXC HA but HAOS VM (from tteck script also)

**Reply by tteck** [reply]
This is not compatible with HAOS as it is already built-in to the file system.

**Comment by shreyasajj** [comment]
Can we use the inbuild bluetooth adapater as a usbpassthroug?
